pkg/services/serviceaccounts/database/token_store.go

```diff
@@ -146,3 +149,31 @@ func (s *ServiceAccountsStoreImpl) detachApiKeyFromServiceAccount(sess *db.Sessi
 
 	return nil
 }
+
+func (s *ServiceAccountsStoreImpl) UpdateAPIKeysExpiryDate(ctx context.Context, expiryDaysLimit int) error {
+	saTokens, err := s.ListTokens(ctx, &serviceaccounts.GetSATokensQuery{})
```

Comment:
This seems like a bit expensive. Could we only list tokens with a expiration time longer than limit and perform updates on them instead of always list all tokens and check them?

Reply:
I've addressed this by extending the existing SA tokens query model. Please take a look at it. Let me know what you think.